src/middlewared/middlewared/main.py

```diff
@@ -380,9 +380,9 @@ def read_write():
         try:
             data = json.loads(form['data'])
             job = await self.middleware.call(data['method'], *(data.get('params') or []))
-        except Exception:
+        except Exception as e:
             resp = web.Response()
-            resp.set_status(405)
+            resp.set_status(405, reason=str(e))
```

Comment:
«Method not allowed» is bad HTTP code for this situation.
For Exception in `json.loads` we should raise 400
If `data['method']` is not present we should raise 422
If corresponding method in middleware is not present we also should raise 422
For Exception in `middleware.call` we should raise 500 (?)

Reply:
Agreed, I will change that.